privacy_consent/models/mail_mail.py

```diff
@@ -10,25 +10,25 @@ class MailMail(models.Model):
     def _postprocess_sent_message(self, success_pids, failure_reason=False,
                                   failure_type=None):
         """Write consent status after sending message."""
-        mail_sent = not failure_type
-        if mail_sent:
-            # Get all mails sent to consents
-            consent_mails = self.filtered(
-                lambda one: one.mail_message_id.model == "privacy.consent"
-            )
+        # Get all mails sent to consents
+        consent_mails = self.filtered(
+            lambda one: one.mail_message_id.model == "privacy.consent"
+        )
+        if consent_mails and success_pids:
             # Get related draft consents
-            consents = self.env["privacy.consent"].browse(
-                consent_mails.mapped("mail_message_id.res_id"),
-                self._prefetch
-            ).filtered(lambda one: one.state == "draft")
+            consents = self.env["privacy.consent"].search([
```

Comment:
Isn't better to filter `consents`?:
```suggestion
            partner_ids = [partner.id for partner in success_pids]
            consents = consents.filtered(lambda x: x.partner_id.id in partner_ids) 
```

Reply:
Before, it was indeed doing a browse with shared prefetch, which made sense because it was being executed after every sent batch, and possibly all consents were already fetched.

Now we only run this code if there's actually consent-related successful mails, and thinking on big mail batches, using filtered might have exponential complexity, while issuing one single query has a constant complexity.

I'm talking theoretically here, I don't have performance analysis of this part. I don't think it's worth it at this point.

Reply:
It turns out this is called per mail, so all of this wasn't really needed. Updated & simplified.